### DISCUSSION TITLE
Webhook is not triggered when creating a new post

### DISCUSSION THREAD

**Discussion by asmbu** [getfider/fider#1029] [Q&A]
Hello,

I am adding and testing webhook with Slack. When I click test, it send http request succesfully and post a new message to my slack channel. But, even tough my webhook is tested and enabled, whenever I create a new post on Fider, Slack does not receive a new message. 

Is there any troubleshooting tips that I can follow? Is there any log that can show me wether my webhook gets triggered or not?

**Comment by goenning** [comment]
When you look at `webhooks` table, is the status = 1 (enabled) and type = 1 (new_post)?

@indyteo any idea what could be happening here?

**Reply by indyteo** [reply]
Can you please share your webhook content template (not the URL obviously) to help debugging this issue?

And the post you created to test it, as well as the logs output to see if something broken is visible.

It should be the same code executed in the test and the real trigger, so can you tell me if the webhook is marked as "failed" after you go back in the admin area after trying it for real, or not?

Oh and can you check the version of Fider you are using? It should be the latest v0.19.1.

**Reply by asmbu** [reply]
@goenning Yes, status is enabled and type is equal to new post.

@indyteo My webhook table
![webhook](https://user-images.githubusercontent.com/74664420/143546548-d6bdfeea-cdde-440e-a809-09d7b9971fc1.png)

The post I created
![new post](https://user-images.githubusercontent.com/74664420/143547590-088403ae-10e2-42a5-a963-129c4c34d042.png)

In the log I can only find entries with "Slack Webhook" generated by testing. When I am creating a post there is no errors generated in the log or in the Admin Settings > Webhooks.

![Webhooks](https://user-images.githubusercontent.com/74664420/143604095-58728b9f-7733-479e-8463-0f7ee3aedc0e.png)

The app is hosted in Azure as an app service and it use image name getfider/fider:main so it should have latest version.

**Reply by indyteo** [reply]
Thank you for your reply.

First of all, concerning your content template, you should use the `escape` function to ensure there are no special characters that blows up your JSON result (see [Security Disclaimer](https://getfider.com/docs/using-webhooks) section in the docs). In your case:
```
{
    "text": "New post {{ escape .post_title }} have been made"
}
```

But with your simple example post, it doesn't explain why is doesn't works. According to the [Slack Incoming Webhooks](https://api.slack.com/messaging/webhooks#posting_with_webhooks) documentation, you should specify a `Content-Type` HTTP header with value `application/json` in the request. Maybe Slack is ignoring the request because of that.

It's weird not to have proper outputs in the logs. Can you verify what is your `LOG_LEVEL` environment variable value? Maybe we could have more information with a `DEBUG` level.

This lack of logs make me think the [`NotifyAboutNewPost`](https://github.com/getfider/fider/blob/main/app/tasks/new_post.go) task doesn't get executed at all. Can you confirm you are receiving the email / web notifications for the event? (Use another account to create the post because you won't receive notifications about your own posts)

Oh and one last thing I just spotted. It seems there is a special character in your username, did you try with a full alpha-numeric name to see if this was causing the issue?
